src/components/ColorPicker.tsx

```diff
@@ -42,14 +48,104 @@ const ColorSwatch: FC<{
+    const currentThoughtValue = (!!state.cursor && getThoughtById(state, head(state.cursor))?.value) || ''
+    const themeColor = themeColors(state)
+    // Define the color and background color regex to get the current color of current thought
+    // document.execCommand('foreColor') adds the color attribute with hex and document.execCommand('backColor') adds the background-color attribute with the rgb
+    const colorRegex = /color="#([0-9a-fA-F]{6})"/g
+    const bgColorRegex = /background-color:\s*(rgb\(\d{1,3},\s?\d{1,3},\s?\d{1,3}\))/g
+    const textHexColor = color ? addAlphaToHex(rgbToHex(color)) : undefined
+    const backHexColor = backgroundColor ? addAlphaToHex(rgbToHex(backgroundColor)) : undefined
+    const selectedTextHexColor = cursorStyle?.color ? addAlphaToHex(rgbToHex(cursorStyle?.color)) : undefined
+    const selectedBackHexColor = cursorStyle?.backgroundColor
+      ? addAlphaToHex(rgbToHex(cursorStyle?.backgroundColor))
+      : undefined
+    if (
+      (cursorStyle?.color === undefined && cursorStyle?.backgroundColor === undefined) ||
+      (selectedTextHexColor === '#ccccccff' && selectedBackHexColor === '#333333ff') ||
+      (selectedTextHexColor === addAlphaToHex(rgbToHex(themeColor.fg)) &&
+        selectedBackHexColor === addAlphaToHex(rgbToHex(themeColor.bg)) &&
+        !selection.isThought())
+    ) {
+      const colorMatches = currentThoughtValue.match(colorRegex) || []
+
+      let matchColor, match
+      // Get the colors and background colors used in current thought's value
+      const fgColors: Set<string> = new Set()
+      if (colorMatches) {
+        // colorMatches will be like this : [color="#ee82ee", color="#ff823e"] and match.slice(7, -1) will be #ee82ee
+        // If the thought is colored with many colors, matchColor will be null and if the thought is colored with one color, matchColor will be that color
+        colorMatches.forEach(match => fgColors.add(match.slice(7, -1)))
+        matchColor = fgColors.size > 1 ? null : fgColors.values().next().value
+      }
+
+      const bgColors: Set<string> = new Set()
+      while ((match = bgColorRegex.exec(currentThoughtValue)) !== null) if (match[1]) bgColors.add(match[1])
+      const matchBgColor = bgColors.size > 1 ? null : bgColors.values().next().value
+
+      return !!(
+        (textHexColor && textHexColor === (matchColor && addAlphaToHex(rgbToHex(matchColor)))) ||
+        (backHexColor && backHexColor === (matchBgColor && addAlphaToHex(rgbToHex(matchBgColor))))
+      )
+    }
+    return !!(
+      (textHexColor && textHexColor === selectedTextHexColor) ||
+      (backHexColor && backHexColor === selectedBackHexColor)
+    )
+  })
+
   /** Toggles the text color to the clicked swatch. */
   const toggleTextColor = (e: React.MouseEvent | React.TouchEvent) => {
     // stop toolbar button dip
     e.stopPropagation()
+    e.preventDefault()
+    console.log(color, backgroundColor, colors.bg, colors.fg)
```

Comment:
Make sure to run the linter before doing a git push, or at least check the CI output after.